### ISSUE TITLE
Custom parameters crashing script

### ISSUE THREAD

**Issue by CannyFoxx** [streamio/streamio-ffmpeg#137]
Hi! I updated gem to v 2.1.0 and every time receiving this error:

```
---------- Encoding started at  8:44:02 PM ----------
/home/cannyfoxx/.rvm/gems/ruby-2.3.0/gems/streamio-ffmpeg-2.1.0/lib/ffmpeg/encoding_options.rb:30:in `block in to_a': no implicit conversion of String into Array (TypeError)
	from /home/cannyfoxx/.rvm/gems/ruby-2.3.0/gems/streamio-ffmpeg-2.1.0/lib/ffmpeg/encoding_options.rb:26:in `each'
	from /home/cannyfoxx/.rvm/gems/ruby-2.3.0/gems/streamio-ffmpeg-2.1.0/lib/ffmpeg/encoding_options.rb:26:in `to_a'
	from /home/cannyfoxx/.rvm/gems/ruby-2.3.0/gems/streamio-ffmpeg-2.1.0/lib/ffmpeg/transcoder.rb:39:in `initialize'
	from /home/cannyfoxx/.rvm/gems/ruby-2.3.0/gems/streamio-ffmpeg-2.1.0/lib/ffmpeg/movie.rb:187:in `new'
	from /home/cannyfoxx/.rvm/gems/ruby-2.3.0/gems/streamio-ffmpeg-2.1.0/lib/ffmpeg/movie.rb:187:in `transcode'
	from convert.rb:14:in `block in <main>'
	from convert.rb:9:in `each'
	from convert.rb:9:in `<main>'
```
After deleting parameter `custom: "-map 0:0 -map 0:1"` all fine. On version 2.0.0 all fine with this argument.

**Comment by bikeath1337** [comment]
Thank you for the report. Will look into it.

**Comment by CannyFoxx** [comment]
Hello again! I tried to use master version of gem with last commit. Still not working.

Code:
```
options = { video_codec: "h264_nvenc", custom: "-map 0:0 -map 0:1" }
transcoder_options = { }

FFMPEG::Movie.new("movie.mkv").transcode("encoded/movie.mp4", options, transcoder_options) do |p|
  #...
end
```
Output:
```
["/usr/bin/ffmpeg", "-y", "-i", "movie.mkv", "-vcodec", "h264_nvenc", "-map 0:0 -map 0:1", "encoded/movie.mp4"]
ffmpeg version 3.1.2 Copyright (c) 2000-2016 the FFmpeg developers
# Information about ffmpeg
Unrecognized option 'map 0:0 -map 0:1'.
Error splitting the argument list: Option not found

Errors: no output file created. 

/usr/local/rvm/gems/ruby-2.3.1/gems/streamio-ffmpeg-2.1.0/lib/ffmpeg/transcoder.rb:109:in `validate_output_file': Failed encoding.Errors: no output file created. Full output: ffmpeg version 3.1.2 Copyright (c) 2000-2016 the FFmpeg developers (FFMPEG::Error)
  built with gcc 6.1.1 (GCC) 20160802
# Information about ffmpeg
Unrecognized option 'map 0:0 -map 0:1'.
Error splitting the argument list: Option not found
	from /usr/local/rvm/gems/ruby-2.3.1/gems/streamio-ffmpeg-2.1.0/lib/ffmpeg/transcoder.rb:45:in `run'
	from /usr/local/rvm/gems/ruby-2.3.1/gems/streamio-ffmpeg-2.1.0/lib/ffmpeg/movie.rb:187:in `transcode'
	from convert.rb:16:in `block in <main>'
	from convert.rb:11:in `each'
	from convert.rb:11:in `<main>'
```

**Comment by bikeath1337** [comment]
Hi there, I added a new spec with your options and they all pass, both with FFMepg 2.8.6 and 3.1.2.

You should try to run the  specs to see if they fail on your end, or provide more information.

**Comment by bikeath1337** [comment]
I figured out the problem. To gain better cross platform compatibility, we stopped using the shell. This meant we didn't need to escape filename, which resulted in several issues. The new approach lets `popen3` deal with the parameters and it absolutely requires that the resulting parameter list look like so:

```ruby
[
'-vcodec',
'h264_nvenc', 
'-map', 
'0:0', 
'-map', 
'0:1'
]
```
There are probably several ways to deal with this but they will involve breaking with the past for the `custom` option, I believe.

Because the custom parameters can contain repeated options (`map`), I _think_ the format for the fix to the `custom` option must be a list because parsing the custom string into a list would be hairy to say the least.

Since the `custom` option is broken anyway, I am leaning towards replacing it with a new option that achieves the same effect. That option would require the user to provide a list of options rather than a string.

Thoughts?
